10/architectures

```diff
@@ -1,8 +1,8 @@
 bashbrew-arch   variants
+amd64    jessie,jessie-slim,alpine,stretch,stretch-slim
 arm32v6  alpine
-arm32v7  jessie,jessie-slim,onbuild,stretch,stretch-slim
-arm64v8  alpine,onbuild,stretch
-amd64    jessie,jessie-slim,alpine,onbuild,stretch,stretch-slim
+arm32v7  jessie,jessie-slim,stretch,stretch-slim
+arm64v8  alpine,stretch,stretch-slim
 i386     alpine
```

Comment:
Why is this one different than all the other version?

Reply:
If you check out https://nodejs.org/dist/latest-v10.x/, there's no `x86` artifact anymore (https://github.com/nodejs/docker-node/pull/823), so only Alpine is supported there (since it builds from source).

Reply:
Yeah, looking at the output of this:

```
 curl -Ss https://nodejs.org/dist/index.json | json -c 'this.version.includes("v10")' | json -ag version files
```

I don't see it either for v10

Reply:
Ok, good to know!!

Reply:
Just to make sure, @MylesBorins @rvagg is this intentional?

Reply:
/cc @nodejs/build

Reply:
I mean, this is just https://github.com/nodejs/build/issues/885, right? (linked from #823 where we had this discussion previously)

Reply:
yeah, no i386 binaries for 10+

Reply:
@rvagg we're building from source for Alpine. Should we stop doing that? Is the plan for Node to not support i386 going forward?

Reply:
Support on Linux for x86 was dropped to experimental with Node 10. The official images probably shouldn't have builds for that since the Node builds no longer officially support it.

Reply:
Easy fix, let me revert this back to where it was in my original push and then any arches Node doesn't push binaries for will go away. :smile:

Reply:
(or should I wait for more consensus here first?)

Reply:
@chorrell @LaurentGoderre @pesho @PeterDaveHello @SimenB @Starefossen (current image maintainers according to `ACTIVE_MAINTAINERS`): should I update this to drop architectures that Node.js does not publish binary artifacts for?

Reply:
We don't even test Node on x86 Linux any more, it's marked "experimental" which means that our infra doesn't test for it, we don't build it but we'll accept community contributions for it. That leaves downstream packagers in an awkward position because you get less support.

I'd suggest that you should drop x86 from Docker. But it'd also be interesting to know what the x86 Docker userbase looks like. Do you have metrics that would indicate what the impact would be?

On another note, I'm pondering setting up an "alternative builds" project that's a little separate to the official Node build infrastructure and nodejs.org offerings. There's a few things that could be offloaded to it, Linux x86 and ARMv6 being the obvious two and Linux musl being a possible addition. We're trying hard to simplify our infra on the official Node end because it's been getting out of hand so setting up something on the side that's unofficial and maybe maintained by an entirely different set of people would be a positive step forward.

Reply:
^ I meant to add that I'd like to gauge interest in such a project and whether there'd be people here willing to contribute time and maybe organisations that might be willing to contribute $$ if needed.

Reply:
I'm working on an "alternative" dist of linux/ia32 for the Internet Archive people. The idea is to have a best-effort-supported dist, and stop when it breaks or becomes intractable.
BTW their use case is embedded systems, so it not obvious to me if that correlates with Docker use cases.

Reply:
> should I update this to drop architectures that Node.js does not publish binary artifacts for?

I assume that means "no binaries at all", there would still be binaries for Alpine even though we don't ship binaries for that on Nodejs.org, Right?

Reply:
Yes, I simply mean dropping things like x86/i386 which are no longer built for officially (even though Alpine images are built from source and could thus still be provided with 32bit builds).